pkg/server/requestLimiter.go

```diff
@@ -25,9 +25,13 @@ func requestLimiterMiddleware(next http.Handler) http.Handler {
 
 		klog.V(6).Info("Checking if we can process incoming request. Current requests: ", len(requestTracker))
 
-		if t, exists := requestTracker[clusterName]; exists {
+		requestTrackerLock.RLock()
+		timeReqReceived, foundClusterProcessing := requestTracker[clusterName]
+		requestTrackerLock.RUnlock()
```

Comment:
So the idea is we are allowing multiple reads here

Reply:
Not exactly, now we are locking while reading the map.
The missing lock was causing a panic if another routine was writing at the same time.

Reply:
Ah okay, I thought we had a lock() there before and it was changed as Rlock().
/lgtm